src/NLog/Config/Factory.cs

```diff
@@ -72,12 +72,13 @@ public void ScanTypes(Type[] types, string prefix)
                 }
                 catch (Exception exception)
                 {
+                    InternalLogger.Error(exception, "Failed to add type '{0}'.", t.FullName);
+                    
                     if (exception.MustBeRethrown())
                     {
                         throw;
                     }
```

Comment:
I think the only reason would be backwardscomp. I think it's bug that the exception was not thrown when `throwExceptions=true`